lib/json_tools.sh

```diff
@@ -5,17 +5,25 @@ flatten_json() {
     # {
     #     "some-amis": {
     #         "us-east-1": "ami-111",
-    #     }
+    #     },
+    #     "periods.in.this.key": 1,
+    # }
+    # into
+    # {
+    #     '["some-amis"].["us-east-1"]': "ami-111",
+    #     '["periods.in.this.key"]': 1,
     # }
-    # into { "some-amis.us-east-1": "ami-111" }
 
     local -r input_json="${1}"
     # https://stackoverflow.com/a/37557003
     jq -r '
+        # Avoids https://github.com/grapl-security/issue-tracker/issues/864
```

Comment:
It may be better to briefly state what this avoids, rather than linking. While this is pretty explicitly a Grapl-only plugin, it _is_ still public, while our issue tracker is not. People could still learn something useful by reading this code, even if they don't work for Grapl.

(or just put that in your commit / PR message)